contrib/99designs/gqlgen/tracer.go

```diff
@@ -0,0 +1,96 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+// Package gqlgen contains an implementation of a gqlgen tracer, and functions to construct and configure the tracer.
+// The tracer can be passed to the gqlgen handler (see package github.com/99designs/gqlgen/handler)
+package gqlgen
+
+import (
+	"context"
+	"fmt"
+	"math"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+
+	"github.com/99designs/gqlgen/graphql"
+)
+
+type trace struct {
+	graphql.NopTracer
```

Comment:
What is the purpose of this? Also note that this is exported, thus becomes public API, potentially callable by users as field `NopTracer`. I can't see it used anywhere at a quick glance.

Reply:
This lets us keep the default implementations of other tracer methods so we don't have to have them in our package. This will not be public because the type itself is private.

Reply:
I see. Can we remove it and just have no-op methods instead? We want to be explicit and clear about the implementation. No surprises, no tricks. This seems like the wrong reason to use embeds. If you want to ensure that the interface is implemented we could (and should) add a test for it, containing only one line:
```go
var _ graphql.Tracer = (*gqlTracer)(nil)
```